shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs

```diff
@@ -751,11 +751,192 @@ mod collections {
+                        i,
+                        get_type_str(*col_type)
+                    )
+                    .as_str()
+                );
+            }
+        }
+
+        fn insert(session: &Session) {
+            // insert maps of native types
+            for (i, col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                let query = format!(
+                    "INSERT INTO test_collections_keyspace.test_map_table_{} (id, my_map) VALUES ({}, {});",
+                    i,
+                    i,
+                    get_map_example(get_type_example(*col_type))
+                );
+                run_query(session, query.as_str());
+            }
+
+            // test inserting map of sets
+            for (i, native_col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                run_query(
+                    session,
+                    format!(
+                        "INSERT INTO test_collections_keyspace.test_map_table_{}_0 (id, my_map) VALUES ({}, {});",
+                        i,
+                        i,
+                        get_map_example(format!("{{{}}}", get_type_example(*native_col_type)).as_str())
+                    )
+                    .as_str()
+                );
+            }
+
+            // test inserting map of lists
+            for (i, native_col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                run_query(
+                    session,
+                    format!(
+                        "INSERT INTO test_collections_keyspace.test_map_table_{}_1 (id, my_map) VALUES ({}, {});",
+                        i,
+                        i,
+                        get_map_example(format!("[{}]", get_type_example(*native_col_type)).as_str())
+                    )
+                    .as_str()
+                );
+            }
+
+            // test inserting map of maps
+            for (i, native_col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                run_query(
+                    session,
+                    format!(
+                        "INSERT INTO test_collections_keyspace.test_map_table_map_{} (id, my_map) VALUES ({}, {{0: {}}});",
+                        i,
+                        i,
+                        get_map_example(get_type_example(*native_col_type))
+                    )
+                    .as_str()
+                );
+            }
+        }
+
+        fn select(session: &Session) {
+            // select sets of native types
+            for (i, col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                let query = format!(
+                    "SELECT my_map FROM test_collections_keyspace.test_map_table_{};",
+                    i
+                );
+
+                assert_query_result(
+                    session,
+                    query.as_str(),
+                    &[&[ResultValue::Map(vec![(
+                        ResultValue::Int(0),
```

Comment:
Every test case has one key-value pair and  the key is always 0 should we be testing more complicated cases?

Reply:
Possibly? It's the same as the set and list tests as they only have one element.

Reply:
ah true, we can consider that for a future improvement then